realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -1520,6 +1630,19 @@ public void endsWith_nullForNullableStrings() {
                 (String) null).findFirst().getFieldStringNotNull());
     }
 
+    // Querying nullable field with like - nulls do not match either '?' or '*'
```

Comment:
Weird... beeender had trouble with this too.  `populateTestRealmForNullTests` populates the DB with 3 rows, one of which is all nulls.  '*' matches two of the rows (not the one with null in it) and '?' matches none of them.  I think the test is pretty good.  Perhaps a comment?